Modules/posixmodule.c

```diff
@@ -2634,13 +2634,14 @@ _posix_free(void *module)
 static PyObject *
 stat_nanosecond_timestamp(_posixstate *state, time_t sec, unsigned long nsec)
 {
-#if SIZEOF_LONG >= 8
+#if SIZEOF_TIME_T == 4
+    return PyLong_FromLongLong(sec * SEC_TO_NS + nsec);
```

Comment:
What is the type of SEC_TO_NS? Should not it or `sec` be cast to `long long`?

We will get a warning also in the very unlikely case of more than 64-bit `long long`. Maybe use condition `SIZEOF_TIME_T + 4 <= SIZEOF_LONG_LONG`?

Reply:
> What is the type of SEC_TO_NS?

It's a long long: it uses the `LL` suffix.

Reply:
> If long long is 128-bit and time_t is 64-bit

Oh ok. Well, I expect many warnings if long long is larger than 64-bit, not just on these lines. We can revisit the code once such platform will exist :-)